scripts/scan-pdf-content.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3
+"""
+Scan user uploaded PDFs in S3 and flag ones which contain any
+non-text or image content (ie. JS or videos).
+
+Requires a running version of veraPDF-REST (https://github.com/veraPDF/veraPDF-rest)
+
+Set AWS_PROFILE to the relevant profile for the stage when running eg.
+$ AWS_PROFILE=development-developer ./scripts/scan-pdf-content.py preview g-cloud-12 http://localhost:8080
+
+Usage: scan-pdf-content.py <stage> <framework> <verapdf-url> [--index=<index>]
+
+Options:
+    <stage>                         Stage to target
+    <framework>                     Slug for the framework to scan
+    <verapdf-url>                   The url for the verapdf-rest service to use
+    --index=<index>                 Optional. If supplied, the scan resumes from a previous run, using
+                                        the list of PDFs stored in '/tmp/pdf-scan-queue.json' starting at `index`
+
+    -h, --help                      Show this information
+
+Examples:
+    ./scripts/scan-pdf-content.py staging g-cloud-12 http://localhost:8080
+"""
+import boto3
+import requests
+import csv
+import logging
+import json
+import os
+import threading
+import queue
+from docopt import docopt
+from datetime import datetime
+
+from typing import Tuple, NamedTuple, Optional
+
+
+class ScanResult(NamedTuple):
+    scanned_at: str
+    status_code: int
+    message: str
+    bucket: str
+    framework: str
+    key: str
+
+
+def list_pdfs_in_bucket(bucket_name: str, framework: str,) -> list:
+    """
+    Return a list of `ObjectSummary`s for PDFs in the supplied S3 bucket.
+
+    :param bucket_name: The name of the bucket to check
+    :param framework: The name of the framework to scan for
+    :return: A list of S3 objects
+    """
+    s3 = boto3.resource("s3")
```

Comment:
Do we need to include the region name here? 🤔 Genuinely not sure.

Reply:
It must pick up the region from the auth & profile you use, there's nothing in [the docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#service-resource) about region and the script runs fine

Reply:
Awesome 👍